minimal multilib build fails

Building current `stable/v2.40` for Arch Linux multilib package (`lib32-util-linux`) fails.

```
[241/534] Compiling C object libmount/python/pylibmount.cpython-312-x86_64-linux-gnu.so.p/pylibmount.c.o
FAILED: libmount/python/pylibmount.cpython-312-x86_64-linux-gnu.so.p/pylibmount.c.o 
gcc -m32 -Ilibmount/python/pylibmount.cpython-312-x86_64-linux-gnu.so.p -Ilibmount/python -I../util-linux/libmount/python -Iinclude -I../util-linux/include -Ilibmount -I../util-linux/libmount -I/usr/include/python3.12 -fvisibility=hidden -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -D_GNU_SOURCE -include /build/lib32-util-linux/src/build/config.h -fno-common -Waddress-of-packed-member -Wdiscarded-qualifiers -Wformat-security -Wimplicit-function-declaration -Wmissing-declarations -Wmissing-parameter-type -Wmissing-prototypes -Wnested-externs -Wno-missing-field-initializers -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wsign-compare -Wstrict-prototypes -Wtype-limits -Wuninitialized -Wunused-but-set-parameter -Wunused-but-set-variable -Wunused-parameter -Wunused-result -Wunused-variable -Wvla -Walloca -D_TIME_BITS=64 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -g -ffile-prefix-map=/build/lib32-util-linux/src=/usr/src/debug/lib32-util-linux -flto=auto -fPIC -Wno-cast-function-type -Wno-error=redundant-decls -MD -MQ libmount/python/pylibmount.cpython-312-x86_64-linux-gnu.so.p/pylibmount.c.o -MF libmount/python/pylibmount.cpython-312-x86_64-linux-gnu.so.p/pylibmount.c.o.d -o libmount/python/pylibmount.cpython-312-x86_64-linux-gnu.so.p/pylibmount.c.o -c ../util-linux/libmount/python/pylibmount.c
In file included from /usr/include/python3.12/Python.h:38,
                 from ../util-linux/libmount/python/pylibmount.h:4,
                 from ../util-linux/libmount/python/pylibmount.c:21:
/usr/include/python3.12/pyport.h:586:2: error: #error "LONG_BIT definition appears wrong for platform (bad gcc/glibc config?)."
  586 | #error "LONG_BIT definition appears wrong for platform (bad gcc/glibc config?)."
      |  ^~~~~
```

This was caused by commit 79546fc1b084839e635eba6736cd809bdbda5d6e, specifically dropping the code from `libmount/python/meson.build`. Adding these lines back makes the build succeed.

The build is very minimal, as I am interested in the libraries only. Please see the [PKGBUILD](https://gitlab.archlinux.org/archlinux/packaging/packages/lib32-util-linux/-/blob/main/PKGBUILD) for the build options used.

--- a/libmount/python/meson.build
+++ b/libmount/python/meson.build
@@ -13,6 +13,11 @@ if LINUX
   pylibmount_sources += 'context.c'
 endif
 
+python_module = import('python')
+python = python_module.find_installation(
+    get_option('python'),
+    required : true,
+    disabler : true)
 python.extension_module(
   'pylibmount',
   pylibmount_sources,

--- a/libmount/python/meson.build
+++ b/libmount/python/meson.build
@@ -1,4 +1,6 @@
-build_python = python.found()
+if get_option('build-python').disabled()
+  subdir_done()
+endif
 
 pylibmount_sources = '''
   pylibmount.c
@@ -11,24 +13,22 @@ if LINUX
   pylibmount_sources += 'context.c'
 endif
 
-if build_python
-  python.extension_module(
-    'pylibmount',
-    pylibmount_sources,
-    include_directories : [dir_include],
-    subdir : 'libmount',
-    dependencies : [mount_dep, python.dependency(embed: true)],
-    c_args : [
-      '-Wno-cast-function-type',
+python.extension_module(
+  'pylibmount',
+  pylibmount_sources,
+  include_directories : [dir_include],
+  subdir : 'libmount',
+  dependencies : [mount_dep, python.dependency(embed: true)],
+  c_args : [
+    '-Wno-cast-function-type',
 
-      # https://github.com/util-linux/util-linux/issues/2366
-      python.language_version().version_compare('>=3.12') ?
-        [ '-Wno-error=redundant-decls' ] : [],
-    ],
-    install : true)
+    # https://github.com/util-linux/util-linux/issues/2366
+    python.language_version().version_compare('>=3.12') ?
+      [ '-Wno-error=redundant-decls' ] : [],
+  ],
+  install : true)
 
-  python.install_sources(
-    '__init__.py',
-    subdir : 'libmount',
-    pure : false)
-endif
+python.install_sources(
+  '__init__.py',
+  subdir : 'libmount',
+  pure : false)

--- a/libmount/python/meson.build
+++ b/libmount/python/meson.build
@@ -18,6 +18,11 @@ python = python_module.find_installation(
     get_option('python'),
     required : true,
     disabler : true)
+if meson.version().version_compare('<1.4.1')
+  cc.has_header('Python.h',
+    include_directories : include_directories(python.get_path('include')),
+    required : true)
+endif
 python.extension_module(
   'pylibmount',
   pylibmount_sources,

--- a/meson.build
+++ b/meson.build
@@ -913,11 +913,7 @@ bison_gen = generator(
   output : ['@BASENAME@.tab.c', '@BASENAME@.tab.h'],
   arguments : ['@INPUT@', '--defines=@OUTPUT1@', '--output=@OUTPUT0@'])
 
-python_module = import('python')
-python = python_module.find_installation(
-    get_option('python'),
-    required : true,
-    disabler : true)
+python_program = find_program('python3', 'python', native : true)
 
 meson_make_symlink = meson.current_source_dir() + '/tools/meson-make-symlink.sh'
 meson_make_manpage_stub = meson.current_source_dir() + '/tools/meson-make-manpage-stub.sh'
@@ -3784,7 +3780,7 @@ configure_file(output : 'meson.conf',
                           '''asan=@0@
 PYTHON=@1@
 '''.format(get_option('b_sanitize')=='address' ? 'yes' : '',
-           python.full_path())])
+           python_program.full_path())])
 
 run_sh = find_program('tests/run.sh')
 run_target(

